src/frequenz/sdk/timeseries/_ringbuffer/buffer.py

```diff
@@ -152,6 +152,24 @@ def update(self, sample: Sample) -> None:
 
         self._update_gaps(timestamp, prev_newest, sample.value is None)
 
+    def time_bound_oldest(self) -> datetime:
+        """
+        Return the time bounds of the ring buffer.
+
+        Returns:
+            The timestamp of the oldest sample of the ring buffer.
+        """
+        return self._datetime_oldest
+
+    def time_bound_newest(self) -> datetime:
+        """
+        Return the time bounds of the ring buffer.
+
+        Returns:
+            The timestamp of the newest sample of the ring buffer.
+        """
+        return self._datetime_newest
```

Comment:
These are good candidates for properties:

```suggestion
    @property
    def time_bound_oldest(self) -> datetime:
        """
        Return the time bounds of the ring buffer.

        Returns:
            The timestamp of the oldest sample of the ring buffer.
        """
        return self._datetime_oldest

    @property
    def time_bound_newest(self) -> datetime:
        """
        Return the time bounds of the ring buffer.

        Returns:
            The timestamp of the newest sample of the ring buffer.
        """
        return self._datetime_newest
```